doc/api/esm.md

```diff
@@ -942,31 +879,22 @@ export function resolve(specifier, context, defaultResolve) {
   return defaultResolve(specifier, context, defaultResolve);
 }
 
-export function getFormat(url, context, defaultGetFormat) {
+export async function load(url, context, defaultLoad) {
   // Now that we patched resolve to let CoffeeScript URLs through, we need to
   // tell Node.js what format such URLs should be interpreted as. For the
   // purposes of this loader, all CoffeeScript URLs are ES modules.
   if (extensionsRegex.test(url)) {
-    return {
-      format: 'module'
-    };
-  }
-
-  // Let Node.js handle all other URLs.
-  return defaultGetFormat(url, context, defaultGetFormat);
-}
-
-export function transformSource(source, context, defaultTransformSource) {
-  const { url, format } = context;
+    const source = await readFile(url);
+    const transformed = CoffeeScript.compile(source, { bare: true });
 
-  if (extensionsRegex.test(url)) {
     return {
-      source: CoffeeScript.compile(source, { bare: true })
+      format: 'module',
```

Comment:
What happens if both `resolve` and `load` return `format`? What happens if they return _different values_ for `format`?

Reply:
I (just now) added a quick addition to `resolve`'s doc above:

```diff
-and optionally its format (such as `'module'`).
+and optionally its format (such as `'module'`) as a hint to the
+`load` hook.
```

Load's `format` is the final word, regardless of what `resolve` did or didn't provide.

Reply:
(Writing quickly based only on what I see in your comment, not on re-reviewing the PR): I think we should probably add a sentence or two covering this case, explaining that the only `format` that matters is the one returned by `load`, and if a loader author creates _only_ a `resolve` hook, and it returns `format`, then they also need to create a `load` hook that might do nothing but return that format along with the default `load`‘s source. As in, if you’re a loader author and you figure out the format in `resolve`, you still need to make sure this format gets returned by `load`.

Should we perhaps make this automatic in Node’s default `load`, that if it sees a format returned by `resolve` then it knows to pass it through as part of the return value of Node’s `load`?

Reply:
Mm, I like both ideas—additional explanation in the docs and automatically handling in node's `defaultLoad` when `resolve` provided a `format` (otherwise, it's a bit of nuisance having to create a custom `load` hook merely to connect what was already done). For the latter, it's an additive change, so it shouldn't break existing tests.

Reply:
Per 25 June meeting, include this as a follow-up.